protocol/transport.go

```diff
@@ -75,8 +82,15 @@ func (t *Transport) endTransaction() (err error) {
 // Read expects to be called only after a call to StartUp without an error response
 // otherwise, an error is returned
 func (t *Transport) Read() (msg Message, err error) {
+	return t.read()
+}
+
+// NextFrontendMessage reads and returns a single message from the connection.
+// NextFrontendMessage expects to be called only after a call to StartUp without an error response
+// otherwise, an error is returned
+func (t *Transport) NextFrontendMessage() (msg pgproto3.FrontendMessage, err error) {
 	if t.transaction != nil {
```

Comment:
should this be the first condition?
are we allow transaction before initialization?

to behave more similar to comment description, what do you think about:
```go
		if !t.initialized {
			err = fmt.Errorf("transport not yet initialized")
			return
		}
		if t.transaction != nil {
			...
		} else {
			...
		}